content-scripts/pages/components_page.tsx

```diff
@@ -0,0 +1,112 @@
+import Icon from "../components/Icon";
+
+const components = ["Icon", "Button", "Table"];
+
+export function createComponentsPage() {
+  if (!document.location.href.toLowerCase().includes("components")) {
+    return;
+  }
+  document.body.innerHTML = "";
+  document.title = "Components";
+
+  const sidebarItems = components.map((component) => (
+    <a href={`#${component}`} key={component}>
+      <li className="se-sidebar-item">
+        {component}
+      </li>
+    </a>
+  ));
+
+  const page = (
+    <div className="se-docs-container">
+      {/* Sidebar */}
+      <div className="se-sidebar">
+        <ul>{sidebarItems}</ul>
+      </div>
+
+      {/* Main Content */}
+      <div className="se-main-content">
+        <h1>NitSig Components</h1>
+        {/* Icon */}
+        <Component name="Icon" description="An icon component..."
+          code={`
+            <Icon name="ri-notification-line" />
+            `}
+        >
+          <Icon name="alert" />
+        </Component>
+
+        {/* Button */}
+        <Component name="Button" description="A button that can be clicked..."
+          code={`
+            <Button
+              name="my_button"
+              text="Click me"
+              icon="ri-notification-line"
+              onclick={() => alert("Button was clicked")}
+            />
+            `}
+
+        >
+          <Button title="my_button" icon="alert" onclick={() => alert("Button was clicked")} />
+        </Component>
+
+
+        {/* Table */}
+        <Component name="Table" description="A Tabela..."
+          code={`
+            <Table
+              name="my_table"
+              headers={[
+                ["Component", "Component"],
+                ["Description", "Description"],
+                ["Status", "Status"]
+              ]}
+              data={[
+                ["Button", "A button that can be clicked", "In progress"],
+                ["Input", "A text input field", "Complete"]
+              ]}
+            />
+            `}
+        >
+          <Table
+            name="my_table"
+            headers={[
+              ["Component", "Component"],
+              ["Description", "Description"],
+              ["Status", "Status"]
+            ]}
+            data={[
+              ["Button", "A button that can be clicked", "In progress"],
+              ["Input", "A text input field", "Complete"]
+            ]}
+          />
+        </Component>
+      </div>
+    </div>
+  );
+
+  document.body.appendChild(page);
+}
+
+interface ComponentProps {
+  name: string;
+  description: string;
+  code: string;
+}
+
+const Component: JSX.Component<ComponentProps> = ({ name, description, code }, children) => (
+  <div id={name} className="se-component-section" >
+    <h2>{name}</h2>
+    <p>{description}</p>
+    <div className="se-component-show">
+      {children}
+    </div>
+    <pre className="se-code-block">
+      {code}
+    </pre>
+  </div>
+);
```

Comment:
It might be better to move this to its own separate file.

Reply:
I don't want to add it to components folder since it is an "internal" thing and not to be used on Sigarra. 
Because is just some lines of code, what do you think to stay here?